pkg/provisioner/ironic/ironic.go

```diff
@@ -1336,6 +1336,25 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 		return p.setMaintenanceFlag(ironicNode, true)
 	}
 
+	if p.host.Status.PoweredOn {
+		p.log.Info("host is being powered off before removal")
+		provResult, err := p.hardPowerOff()
+		switch {
+		case err == nil:
+			p.host.Status.PoweredOn = false
```

Comment:
Don't we want to wait for ironic to actually power the host off? That might take a few seconds, during which we need to keep trying. If we set PoweredOn to false here, then the check on 1339 will fail and we'll delete the host on the next pass through.

Reply:
I think this is fixed also.